Data/Repositories/RecruitmentRepository.cs

```diff
@@ -1,36 +1,76 @@
+using AutoMapper;
 using Common.Enums;
 using Data.Entities;
 using Microsoft.Data.SqlClient;
+using Data.DTOs.Recruitment;
 using System.Security.Claims;
+using Data.DTOs.RecruitmentSkill;
 
 namespace Data.Repositories
 {
     public class RecruitmentRepository : BaseRepository<Recruitment>
     {
-        public RecruitmentRepository(DataContext context) : base(context)
+        private readonly IMapper _mapper;
+        private readonly RecruitmentSkillRepository _recruitmentSkillRepository;
+        public RecruitmentRepository(DataContext context, IMapper mapper, RecruitmentSkillRepository repo) : base(context)
         {
-            
+            _mapper = mapper;
+            _recruitmentSkillRepository = repo;
         }
-
-        public int GetRecruiterId(int id)
+        public RecruitmentDetailsDTO GetRecruitmentById(int id)
         {
             Recruitment recruitment = GetById(id);
-            int result = recruitment.RecruiterId;
+
+            if(recruitment==null)
+            {
+                return null;
+            }
+            if (recruitment.DeletedDate.HasValue)
+            {
+                return null;
+            }
+
+            RecruitmentDetailsDTO result = _mapper.Map<RecruitmentDetailsDTO>(recruitment);
+            result.Skills = _recruitmentSkillRepository.GetAllRecruitmentSkills(result.Id);
```

Comment:
You can't use repository in repository

Reply:
You told me to get Recruitment objects with the Skills from the repository. How can I do that without using the repository for the RecruitmentSkill table?

Reply:
Should I just use DataContext there and delete the repository for the RecruitmentSkill table?